src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -115,11 +121,35 @@ private void ParseRequestLine(TRequestHandler handler, ReadOnlySpan<byte> reques
             // Consume space
             offset++;
 
-            // Version + CR is 9 bytes which should take us to .Length
-            // LF should have been dropped prior to method call
-            if ((uint)offset + 9 != (uint)requestLine.Length || requestLine[offset + sizeof(ulong)] != ByteCR)
+            if (!_enableLineFeedTerminator)
             {
-                RejectRequestLine(requestLine);
+                // Version + CR is 9 bytes which should take us to .Length
+                // LF should have been dropped prior to method call
+                if ((uint)offset + 9 != (uint)requestLine.Length || requestLine[offset + 8] != ByteCR)
+                {
+                    RejectRequestLine(requestLine);
+                }
+            }
+            else
+            {
+                // LF should have been dropped prior to method call
+                // If offset + 8 is .Length then requestLine is valid since it mean LF was the next char
+                if ((uint)offset + 8 != (uint)requestLine.Length)
+                {
+                    // Version + CR is 9 bytes which should take us to .Length
+                    if ((uint)offset + 9 != (uint)requestLine.Length || requestLine[offset + 8] != ByteCR)
+                    {
+                        RejectRequestLine(requestLine);
+                    }
+                }
+                else 
+                {
+                    // e.g., GET / HTTP1.\r\n
+                    if (requestLine[offset + 7] == ByteCR)
```

Comment:
Doesn't `remaining.GetKnownVersion()` already return `HttpVersion.Unknown` in this case?

Reply:
Could this be simpler?

```csharp
// Version + CR is 9 bytes which should take us to .Length
// LF should have been dropped prior to method call
if ((uint)offset + 9 != (uint)requestLine.Length || requestLine[offset + 8] != ByteCR)
{
    // LF should have been dropped prior to method call
    // If _enableLineFeedTerminator and offset + 8 is .Length,
    // then requestLine is valid since it mean LF was the next char
    if (!_enableLineFeedTerminator || (uint)offset + 8 != (uint)requestLine.Length)
    {
         RejectRequestLine(requestLine);
    }
}
```

Reply:
With your suggestion this is failing this test

```
Expected: Invalid request line: 'GET / HTTP/1.\x0D'
Actual:   Unrecognized HTTP version: 'HTTP/1.'
```

Reply:
In theory it could be fine since it fails as expected, but with functional tests it shows that the resulting status code is different:

```
| Expected: HTTP/1.1 400 Bad Request\r\nContent-Length: 0\r\nConne···
| Actual:   HTTP/1.1 505 HTTP Version Not Supported\r\nContent-L···
```

Reply:
We could even assume that `HTTP/1.` is actually a wrong version, and not that the request is bad. Updating the test to reflect that. That would be considered a breaking change, right?

Reply:
I'm fine with returning a 505 in this case if that simplifies the code.

Reply:
But then `HTTP/1.` return 505 while `HTTP/1`, `HTTP/`, ... would return 400. Still ok?